app/models/authentication.rb

```diff
@@ -11,4 +11,12 @@ class Authentication < ApplicationRecord
 
   attribute :availability_status, :string
   validates :availability_status, :inclusion => { :in => %w[available unavailable] }, :allow_nil => true
+
+  validate :authtype_not_updated
+
+  def authtype_not_updated
+    if !new_record? && authtype_changed?
+      errors.add(:authtype, "cannot be updated")
```

Comment:
@slemrmartin You can add `on: :update` to the `validate` line then the condition here is very simple:

```ruby
validate :authtype_not_updated, :on => :update

def authtype_not_updated
  errors.add(:authtype, "cannot be updated") if authtype_changed?
end
```

Reply:
@gmcculloug I was thinking about it, on one side, it makes the "validation" line more obvious, on the other side, it makes the validation method less obvious so current version seems easier to understand to me (it's more important when the class becomes bigger)